src/main/java/com/minecolonies/colony/buildings/BuildingGuardTower.java

```diff
@@ -192,7 +536,29 @@ public View(ColonyView c, BlockPos l)
         @Override
         public com.blockout.views.Window getWindow()
         {
-            return new WindowHutWorkerPlaceholder<>(this, GUARD_TOWER);
+            return new WindowHutGuardTower(this);
+        }
+
+        @Override
+        public void deserialize(@NotNull final ByteBuf buf)
+        {
+            super.deserialize(buf);
+            assignManually = buf.readBoolean();
+            retrieveOnLowHealth = buf.readBoolean();
+            patrolManually = buf.readBoolean();
+            task = Task.values()[buf.readInt()];
+            int jobId = buf.readInt();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'jobId' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)